lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -0,0 +1,614 @@
+           llvm::function_ref<void(ArrayAttr)> addToWorklist) {
+  return success();
+}
+
+static LogicalResult applyWithoutTargetImpl(AnnoPathValue target,
+                                            DictionaryAttr anno,
+                                            applyState state,
+                                            bool allowNonLocal) {
+  if (!allowNonLocal && !target.isLocal())
+    return failure();
+  SmallVector<NamedAttribute> newAnnoAttrs;
+  for (auto &na : anno) {
+    if (na.first != "target") {
+      newAnnoAttrs.push_back(na);
+    } else if (!target.isLocal()) {
+      auto sym = scatterNonLocalPath(target, state);
+      newAnnoAttrs.push_back(
+          {Identifier::get("circt.nonlocal", anno.getContext()), sym});
+    }
+  }
+  addAnnotation(target.ref, newAnnoAttrs);
+  return success();
+}
+
+template <bool allowNonLocal, typename T, typename... Tr>
+static LogicalResult applyWithoutTarget(AnnoPathValue target,
+                                        DictionaryAttr anno, applyState state) {
+  if (!target.isOpOfType<T, Tr...>())
+    return failure();
+  return applyWithoutTargetImpl(target, anno, state, allowNonLocal);
+}
+
+template <bool allowNonLocal = false>
+static LogicalResult applyWithoutTarget(AnnoPathValue target,
+                                        DictionaryAttr anno, applyState state) {
+  return applyWithoutTargetImpl(target, anno, state, allowNonLocal);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Driving table
+////////////////////////////////////////////////////////////////////////////////
+
+namespace {
+struct AnnoRecord {
+  llvm::function_ref<Optional<AnnoPathValue>(DictionaryAttr, applyState)>
+      resolver;
+  llvm::function_ref<LogicalResult(AnnoPathValue, DictionaryAttr, applyState)>
+      applier;
+};
+}; // namespace
+
+static const llvm::StringMap<AnnoRecord> annotationRecords{{
+
+    // Testing Annotation
+    {"circt.test", {stdResolve, applyWithoutTarget<true>}},
+    {"circt.testNT", {noResolve, applyWithoutTarget<>}},
+    {"circt.missing", {tryResolve, applyWithoutTarget<>}}
+
+}};
+
+static const AnnoRecord *getAnnotationHandler(StringRef annoStr,
+                                              bool ignoreUnhandledAnno) {
+  auto ii = annotationRecords.find(annoStr);
+  if (ii != annotationRecords.end())
+    return &ii->second;
+  if (ignoreUnhandledAnno)
+    return &annotationRecords.find("circt.missing")->second;
+  return nullptr;
+}
+
+//===----------------------------------------------------------------------===//
+// Pass Infrastructure
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct LowerAnnotationsPass
+    : public LowerFIRRTLAnnotationsBase<LowerAnnotationsPass> {
+  void runOnOperation() override;
+  LogicalResult applyAnnotation(DictionaryAttr anno, applyState state);
+
+  bool ignoreUnhandledAnno = false;
+  bool ignoreClasslessAnno = false;
```

Comment:
I like this classless infra. (I think this is new?) I like writing classless annotations for testing, but these should not exist coming from Chisel. Having a mechanism to enable/disable this behavior makes sense to me. 💯